documentation/sdkautomation/sdk_customization.md

```diff
@@ -0,0 +1,762 @@
+            "const": false
+          }
+        ],
+        "default": false
+      },
+      "installInstructionScript": {
+        // Generate install instruction that could be shown in spec PR comment (lite version)
+        //  or in generated SDK PR (full version).
+        // If generate_output.json contains install_instruction then this could be skipped.
+        // Param: <path_to_installInstructionInput.json> <path_to_installInstructionOutput.md> <lite|full>
+        // installInstructionInput.json: See #InstallInstructionScriptInput .
+        // installInstructionOutput.md: Install instruction in markdown.
+        "allOf": {
+          "$ref": "#/definitions/RunOptions"
+        },
+        "properties": {
+          "enableLiteInstallInstruction": {
+            "type": "boolean",
+            "default": false
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RunOptions": {
+      // Options to run a script and collect log.
+      "type": "object",
+      "properties": {
+        "path": {
+          // Script path related to repo root
+          "type": "string"
+        },
+        "logPrefix": {
+          // Prefix to be added to SDK Automation log. By default it would be filename of the script.
+          "type": "string"
+        },
+        "stdout": {
+          // How should SDK Automation handle the script stdout stream
+          "$ref": "#/definitions/RunLogOptions"
+        },
+        "stderr": {
+          // How should SDK Automation handle the script stderr stream
+          "$ref": "#/definitions/RunLogOptions"
+        },
+        "exitCode": {
+          // How should SDK Automation handle non-zero exitCode.
+          "showInComment": {
+            // Should we show this error in comment.
+            "type": "boolean",
+            "default": true
+          },
+          "result": {
+            // If script has non-error exitCode how should we mark the script's result.
+            "type": "string",
+            "enum": [
+              "error", "warning", "ignore"
+            ],
+            "default": "error"
+          }
+        },
+        "required": [
+          "path"
+        ]
+      }
+    },
+    "RunLogOptions": {
+      // How should SDK Automation handle the log stream.
+      "showInComment": {
+        // Should we show this stream in comment.
+        "$ref": "#/definitions/RunLogFilterOptions"
+      },
+      "scriptError": {
+        // If any line match, assume the script fails.
+        "$ref": "#/definitions/RunLogFilterOptions"
+      },
+      "scriptWarning": {
+        // If any line match, assume the script warns.
+        "$ref": "#/definitions/RunLogFilterOptions"
+      }
+    },
+    "RunLogFilterOptions": {
+      "oneOf": [
+        {
+          // If line of log match this regex then hit
+          "type": "string",
+          "format": "regex"
+        },
+        {
+          // If set to true, any line of log will hit
+          "type": "boolean"
```

Comment:
We could consider using only a regex and use `.*` for places where it takes everything.

Reply:
But what if you want it to match nothing?

Reply:
You could put empty or some string that will not match anything.